Description: Backport of fix for a bug that causes crashes with programs
 such as easytag.
 .
 It deserves to be applied in Jessie because it makes several programs
 suddenly crash apparently out of nowhere.
Author: Ruben Undheim <ruben.undheim@gmail.com>
Bug-Debian: https://bugs.debian.org/788002
Origin: upstream
Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1163579
Last-Update: 2015-08-17

Index: gtk+3.0/gtk/gtktreeview.c
===================================================================
--- gtk+3.0.orig/gtk/gtktreeview.c	2015-08-17 09:57:38.613296608 +0200
+++ gtk+3.0/gtk/gtktreeview.c	2015-08-17 10:46:42.893021554 +0200
@@ -5576,8 +5576,8 @@
 
       view_rect.x = 0;
       view_rect.y = gtk_tree_view_get_effective_header_height (tree_view);
-      view_rect.width = gdk_window_get_width (tree_view->priv->bin_window);
-      view_rect.height = gdk_window_get_height (tree_view->priv->bin_window);
+      view_rect.width = gtk_widget_get_allocated_width (widget);
+      view_rect.height = gtk_widget_get_allocated_height (widget) - view_rect.y;
 
       gdk_window_get_position (tree_view->priv->bin_window, &canvas_rect.x, &canvas_rect.y);
       canvas_rect.y = -gtk_adjustment_get_value (tree_view->priv->vadjustment);
